src/components/Campaign.tsx

```diff
@@ -27,12 +28,16 @@ export const Campaign = () => {
   const campaignContentQuery = useGetCampaignContent(idCampaign);
   const campaignContentMutation = useUpdateCampaignContent();
 
-  const { save } = useSingletonEditor(
+  const { save, exportContent, isExporting } = useSingletonEditor(
     {
       initialContent: campaignContentQuery.data,
       onSave: (content: Content) => {
         campaignContentMutation.mutate({ idCampaign, content });
       },
+      onExport: (content: UnlayerContent) => {
+        setContentToExport(content);
+        setOpen(true);
+      },
     },
     [campaignContentQuery.data, campaignContentMutation.mutate, idCampaign]
   );
```

Comment:
```suggestion
  const { save, export, isExporting } = useSingletonEditor(
    {
      initialContent: campaignContentQuery.data,
      onSave: (content: Content) => {
        campaignContentMutation.mutate({ idCampaign, content });
      },
    },
    [campaignContentQuery.data, campaignContentMutation.mutate, idCampaign]
  );

  const exportContent = (content: UnlayerContent) => {
      setContentToExport(content);
      setOpen(true);
  };
```

I am not totally sure, but in my opinion, it looks cleaner and it is more flexible.

Reply:
`save` is different because it is used internally by the `SingletonEditor`.

Reply:
What is more! you do not need `isExporting`: 

```suggestion
  const [ isExportingAstemplate, setIsExportingAstemplate ] = useState(false);

  const { save, export } = useSingletonEditor(
    {
      initialContent: campaignContentQuery.data,
      onSave: (content: Content) => {
        campaignContentMutation.mutate({ idCampaign, content });
      },
    },
    [campaignContentQuery.data, campaignContentMutation.mutate, idCampaign]
  );

  const exportAsTemplate = () => {
    setIsExportingAsTemplate(true);
    try {
      const content = await export();
      setContentToExportAsTemplate(content);
      setExportAsTemplateDialogoOpen(true);
    } finally {
      setIsExportingAsTemplate(false);
    }
  };
```

Reply:
Good approach, was implemented. 👍🏻